dao/database.js

```diff
@@ -43,6 +45,40 @@ function updateVersion() {
   });
 }
 
+function run(...args) {
+  return new Promise((resolve, reject) => {
+    // Cannot use arrow function because `this` is set by the caller
+    db.run(...args, function(err) {
+      if (err) {
+        reject(err);
+        return;
+      }
+      resolve({ lastId: this.lastID, changes: this.changes });
+    });
+  });
+}
+
+function safeUpdateOrDelete(mode, ...args) {
+  args[0] = `${mode.toUpperCase()} ${args[0]}`;
```

Comment:
we don't really care about the uppercase, it's mostly to make things more readable when reading the statement.